Modernize examples (#720)
w3c/webtransport: explainer.md

```javascript
sendParticipant(encodedVideoChunksParticipantA, wt.createSendGroup());
sendParticipant(encodedVideoChunksParticipantB, wt.createSendGroup());

async function sendParticipant(realtimeEncodedVideoChunks, sendGroup) {
  let sendOrder = 0;
  for await (const encodedVideoChunk of realtimeEncodedVideoChunks.readable) {
    const bytes = new Uint8Array(encodedVideoChunk.byteLength);
    encodedVideoChunk.copyTo(bytes);
    const writable = await wt.createUnidirectionalStream({ sendGroup, sendOrder: sendOrder-- });
    const writer = writable.getWriter();
    writer.write(bytes).catch(() => {});
    writer.close();
  }
}
```

### 7. Transactional Writes and Reliable Reset

WebTransport also supports transactional writes and reliable reset via the `atomicWrite()` and `commit()` methods respectively. The former ensures that bytes only go out together, and the latter commits to sending what has been written up to this point even if the stream is later aborted.

```javascript
const writable = await wt.createUnidirectionalStream();
const writer = writable.getWriter();
try {
  await writer.atomicWrite(bytes);
} catch (e) {
  if (e.name != "AbortError") throw e;

>   // Blocked on flow control; the writable remains un-errored.

Reviewer: ```suggestion
  // Write prevented by flow control. The writable remains un-errored.
```

```javascript
const wt = new WebTransport('https://example.com/wt', {
  protocols: ['v2.chat', 'v1.chat'], // Proposed to the server
  requireUnreliable: true            // Fail if only H2/TCP available
});
await wt.ready;
console.log(`The server selected ${wt.protocol || "no"} protocol`); 
console.log(wt.reliability); // supports-unreliable
```

### 2. Connection with Headers and Cert Hashes

```javascript
// Replace these two with real values from your server.
const token = "dev-token-123";
const certHashHex = "edb03e3a0a5fbb4c1c8e62c8a0cf9c54c2e5a6d3b2b4a1c9d0e1f2a3b4c5d6e7";

const wt = new WebTransport("https://127.0.0.1:4433/wt", {
  headers: { Authorization: `Bearer ${token}` },
  serverCertificateHashes: [{ algorithm: "sha-256", value: hexToU8(certHashHex) }],
});
await wt.ready;

function hexToU8(hex) {
  const out = new Uint8Array(hex.length / 2);
  for (let i = 0; i < out.length; i++) out[i] = parseInt(hex.slice(i * 2, i * 2 + 2), 16);
  return out;
}
```

### 3. Unidirectional and Bidirectional Streams

```javascript
// Receiving server-initiated unidirectional streams of data
for await (const readable of wt.incomingUnidirectionalStreams) {

>   // consume streams independently using IFFEs, reporting per-stream errors

Author: redo to expand of IFFE or use sub function

writer.write("Hello Server").catch(() => {});
writer.close();
await encoder.readable.pipeTo(await wt.createUnidirectionalStream());

// Use non-blocking bidirectional streams as a request/response pattern
const encoder = new TextEncoderStream();
const writer = encoder.writable.getWriter();
writer.write("Hello Server").catch(() => {});
writer.close();
await encoder.readable
  .pipeThrough(await wt.createBidirectionalStream())
  .pipeThrough(new TextDecoderStream())
  .pipeTo(new WritableStream({write: msg => console.log(msg)}); // "Hi client"
```

### 4. Sending and Receiving Datagrams

Ideal for high-frequency, time-sensitive data.

```javascript
// Decode server-initiated datagrams into text
const decoder = new TextDecoder();
for await (const datagram of wt.datagrams.readable) {
  console.log(decoder.decode(datagram));
}

// Send encoded datagrams to the server
const writable = wt.datagrams.createWritable();
const writer = writable.getWriter();
const encoder = new TextEncoder();
for (const message of messages) {
  const datagram = encoder.encode(message);
  if (datagram.length > wt.datagrams.maxDatagramSize) throw;
  await writer.ready;

>   writer.write(datagram).catch(() => {});

Reviewer 2: Is this the sort of error handling we would encourage?
